From a7195023c2120973b817866f9d324c39d4f07286 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Wed, 3 Jul 2024 10:24:13 +0200
Subject: [PATCH 1039/1695] core: firewall: fix firewall_dt_get_by_name()

The property to identify an entry in the "access-controllers" property
is "access-controller-names", not "access-controllers-names".

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3de825059896080f373cad1105771cd8308b46b3 core: firewall: fix firewall_dt_get_by_name()]
Change-Id: I3f85fb22367578aa11e5ddcfbddb33ef7c3e261d
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391788
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/firewall/firewall.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/firewall/firewall.c b/core/drivers/firewall/firewall.c
index a2e7d07af..8cec8eb0d 100644
--- a/core/drivers/firewall/firewall.c
+++ b/core/drivers/firewall/firewall.c
@@ -83,7 +83,7 @@ TEE_Result firewall_dt_get_by_name(const void *fdt, int node, const char *name,
 {
 	int index = 0;
 
-	index = fdt_stringlist_search(fdt, node, "access-controllers-names",
+	index = fdt_stringlist_search(fdt, node, "access-controller-names",
 				      name);
 	if (index == -FDT_ERR_NOTFOUND)
 		return TEE_ERROR_ITEM_NOT_FOUND;
-- 
2.47.2

